File: scaffolds/leetcode/_past/1512_number_of_good_pairs/1512_number_of_good_pairs.go
package main

import "fmt"

func numIdenticalPairs(nums []int) int {
    goodPairs := 0

    for i := 0; i < len(nums); i++ {
        for j := i + 1; j < len(nums); j++ {
            if nums[j] == nums[i] {
                goodPairs++
            }
        }
    }

    return goodPairs
}

func main() {
    fmt.Println(numIdenticalPairs([]int{1, 2, 3, 1, 1, 3})) // 4
}

File: scaffolds/leetcode/_past/1512_number_of_good_pairs/1512_number_of_good_pairs.ts
function numIdenticalPairs(nums: number[]): number {
  let goodPairs: number = 0;

  for (let i: number = 0; i < nums.length; i++) {
    for (let j: number = 0; j < nums.length; j++) {
      if (i < j && nums[j] == nums[i])
        goodPairs++;
    }
  }

  return goodPairs;
};

console.log(numIdenticalPairs([1, 2, 3, 1, 1, 3])); // 4
